File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.cpp
const int MOD = 1e9 + 7;
class Solution {
   public:
    int numSub(string s) {
        int n = s.size();
        long long ans = 0;
        int left = 0;
        for (int right = 0; right < n; right++) {
            if (s[right] == '0') {
                left = right + 1;
            } else {
                ans = (ans + (right - left + 1)) % MOD;
            }
        }
        return ans % MOD;
    }
};

File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.rs
impl Solution {
    pub fn num_sub(s: String) -> i32 {
        let bs = s.as_bytes();
        let MOD = 1e9 as i64 + 7;
        let n = s.len();
        let mut left = 0 as usize;
        let mut ans = 0 as i64;

        for right in 0..n {
            if bs[right] == b'0' {
                left = right + 1;
            } else {
                ans = (ans + (right - left + 1) as i64) % MOD;
            }
        }

        ans as i32
    }
}